services/secure-gateway/security/matchResource_test.go

```diff
@@ -0,0 +1,187 @@
+package security
+
+import (
+	"reflect"
+	"testing"
+)
+
+var TestCases = []struct {
+	name      string
+	given     string
+	parsed    map[string]string
+	resource  map[string]string
+	wantMatch bool
+	wantError bool
+}{
+	{
+		name:      "Empty",
+		given:     "",
+		parsed:    map[string]string{},
+		resource:  map[string]string{},
+		wantMatch: true,
+	},
+	{
+		name:      "Single Key",
+		given:     "/path/to/key",
+		parsed:    map[string]string{"": "/path/to/key"},
+		resource:  map[string]string{"": "/path/to/key"},
+		wantMatch: true,
+	},
+	{
+		name:      "Single context property",
+		given:     "property",
+		parsed:    map[string]string{"": "property"},
+		resource:  map[string]string{"": "property"},
+		wantMatch: true,
+	},
+	{
+		name:      "Single context with context property",
+		given:     "user=test:prop",
+		parsed:    map[string]string{"user": "test", "": "prop"},
+		resource:  map[string]string{"user": "test", "": "prop"},
+		wantMatch: true,
+	},
+	{
+		name:      "Two contexts with context prop",
+		given:     "user=test+device=test2:prop",
+		parsed:    map[string]string{"user": "test", "device": "test2", "": "prop"},
+		resource:  map[string]string{"user": "test", "device": "test2", "": "prop"},
+		wantMatch: true,
+	},
+	{
+		name:      "Multiple contexts with context prop",
```

Comment:
We need to figure out what to do with this, we probably need full names for properties

Reply:
since we need to add identity prefix anyway, let's add "keys" prefix as well and it can replace the "/" prefix